python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -328,8 +333,93 @@ def get_trainer_program(self):
         # FIXME(typhoonzero): Also ops like clip_gradient, lrn_decay?
         delete_ops(self.origin_program.global_block(), self.optimize_ops)
         self.origin_program.__str__()
+
         return self.origin_program
 
+    def _get_trainer_startup_program(self, program=None):
+        """
+        Get transpiled trainer side startup program.
+
+        Returns:
+            Program: trainer side startup program.
+        """
+        if program is None:
+            program = self.startup_program
+
+        orig_s_prog = program
+
+        #delete initialize operators.
+        assert (orig_s_prog.num_blocks == 1)
+        # FIXME(gongwb): delete not need ops.
+        # note that: some parameter is not trainable and they ops can't be deleted.
+        # delete_ops(orig_s_prog.global_block(), orig_s_prog.global_block().ops)
+
+        # add concat ops to origin parameters in startup program to
+        # let the origin parameters initialized by the spilited parameters
```

Comment:
add concat ops to merge splited parameters received from parameter servers.

Reply:
Done.